orchestrator/careplancontributor/ehr/notifier.go

```diff
@@ -112,7 +112,10 @@ func (n *notifier) processTaskAcceptedEvent(ctx context.Context, event *TaskAcce
 			log.Ctx(ctx).Warn().Err(err).Msg("Task enrollment failed due to bad request")
 			task := event.Task
 			task.Status = fhir.TaskStatusRejected
-			err = cpsClient.UpdateWithContext(ctx, "Task/"+*event.Task.Id, task, &task)
+			task.StatusReason = &fhir.CodeableConcept{
+				Text: badRequest.Reason,
+			}
+			err = cpsClient.UpdateWithContext(context.Background(), "Task/"+*event.Task.Id, task, &task)
```

Comment:
Could you revert this to `ctx`? I pushed a comment with a potential fix for the issue.

My theory is as follows: this context comes from the "receive" handler of the messaging module. On dev/test, this is Azure ServiceBus. Locally, it is sourced from the caller of `SendMessage()`, but that context ultimately originates from something that a client did (e.g. invoking an HTTP endpoint). So that context might've expired/cancelled by the caller, leading to the message handler receiving an cancelled context. Which is incorrect, because messaging should decouple this (that's why we use it in the first place).

I don't understand why the error occurs on `UpdateWithContext` and not earlier (might be due to timing), but let's see if this fixes it.